lib/index.js

```diff
@@ -2210,6 +2210,27 @@ addMethod('flatMap', function (f) {
     return this.map(f).sequence();
 });
 
+/**
+ * Creates a new Stream of values by applying each item in a Stream to each
+ * value in a Stream of functions
+ *
+ * @id ap
+ * @section Higher-order Streams
+ * @name Stream.ap(m)
+ * @param {Stream} m - incoming stream of values to apply to function(s) in stream
+ * @api public
+ *
+ * var readFile = _.wrapCallback(fs.readFile);
+ * filenames.flatMap(readFile)
+ */
+
+addMethod('ap', function (m) {
+    return this.map(function (f) {
+        return m.fork().map(f);
+    })
+      .parallel(Infinity);
```

Comment:
Typically I would use `.merge()` here because of `m.fork()`, but I feel like that is taking too much liberty with users' code since order of results is not guaranteed. I think that also violates the `composition` specification because the results will not necessarily be the same if the function or values are asynchronous (re: events).